### ISSUE TITLE
Where is pod logs and terminal button

### ISSUE THREAD

**Issue by pydr** [lensapp/lens#1690]
**Describe the bug**
Dear lens develop team:
I updated my lens app to ver 4.0.0.
but I can't find pod logs and terminal button.

**Screenshots**
![download _1_.png](https://i.loli.net/2020/12/08/WpRiex9ubMvYUNC.png)

**Environment (please complete the following information):**
- Lens Version:
- OS: Ubuntu 18.04
- Installation method (e.g. snap or AppImage in Linux): snap

**Comment by aabdullinov** [comment]
Hi, 
I have the same problem about pod logs on windows 10 desktop.
detailed screenshots in attachment.
![lens4 0 1](https://user-images.githubusercontent.com/73327504/101460333-a522d100-3963-11eb-8ccd-6227ddc8d865.PNG)
![lens4 0 2](https://user-images.githubusercontent.com/73327504/101460338-a653fe00-3963-11eb-931a-b1d5d33f7714.PNG)
![lens4 0 3](https://user-images.githubusercontent.com/73327504/101460339-a6ec9480-3963-11eb-9e8c-1373dc8564ed.PNG)
![lens4 0 4](https://user-images.githubusercontent.com/73327504/101460341-a7852b00-3963-11eb-8c30-9c07eeca9050.PNG)
![lens4 0 5](https://user-images.githubusercontent.com/73327504/101460344-a7852b00-3963-11eb-8234-34a6874bd9d5.PNG)

**Comment by nevalla** [comment]
Logs and terminal buttons are now run as extensions and it seems that those are not loaded properly on your environment. If you open Developer tools (View / Toggle Developer Tools), do you see this line in the console:
```
info: [EXTENSION]: enabled lens-pod-menu@0.1.0
```

Or do you see any errors there?

**Comment by pydr** [comment]
Thanks for your reply.
yes, I saw an error in the console:

![download _2_.png](https://i.loli.net/2020/12/08/Nwv9Vba74DJUxGe.png)

**Comment by aabdullinov** [comment]
@nevalla
Thanks for your reply!
Yes, there are a lot of errors, and some of them is about "lens-pod-menu":

```
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\AAbdullinov\AppData\Roaming\Lens\node_modules\lens-pod-menu\dist\renderer.js: Error: Cannot find module 'C:\Users\AAbdullinov\AppData\Roaming\Lens\node_modules\lens-pod-menu\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\AAbdullinov\AppData\Roaming\Lens\node_modules\lens-pod-menu\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:43)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2
```

**Comment by nevalla** [comment]
@pydr if you go into `/home/dev/snap/kontena-lens/144/.config/Lens/node_modules/lens-pod-menu/` do you see `dist` folder and `renderer.js` in `dist`?

**Comment by pydr** [comment]
Hello @nevalla .
I checked this dir, but there is no any file.

`cd /home/dev/snap/kontena-lens/144/.config/Lens/node_modules/lens-pod-menu/
cd: no such file or directory: /home/dev/snap/kontena-lens/144/.config/Lens/node_modules/lens-pod-menu/`

`ll /home/dev/snap/kontena-lens/144/.config/Lens/node_modules
total 0
`

**Comment by jhcook** [comment]
I'm Homebrew and same issue. Just as I was loving it :-( 

```
$ brew info lens
lens: 4.0.0
https://k8slens.dev/
/usr/local/Caskroom/lens/4.0.0 (118B)
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/lens.rb
==> Name
Lens
==> Description
Kubernetes IDE
==> Artifacts
Lens.app (App)
==> Analytics
install: 1,447 (30 days), 5,275 (90 days), 11,380 (365 days)
```

**Comment by stijlbreuk-dev** [comment]
On Mac I fixed it by renaming `/Users/myusername/Library/Application Support/Lens` to _Lens (so we have a backup)
Remove the application completely, redownloading it, reinstalling it.

**Comment by jakolehm** [comment]
Could someone start Lens from command line using `LOG_LEVEL=debug`, see https://docs.k8slens.dev/v4.0.1/faq/#how-can-i-see-more-verbose-logs for more details.

**Comment by pydr** [comment]
Hey @jakolehm .
Thanks for your reply.

I ran lens app with debug mode, and I got some logs:
```
➜  ~ LOG_LEVEL=debug /snap/kontena-lens/145/kontena-lens
info: 🚀 Starting Lens from "/home/dev/.config/Lens"
warn: APPIMAGE env is not defined, current application is not an AppImage
STORE MIGRATION (/home/dev/.config/Lens/config.json): 2.1.0-beta.4
info: [STORE]: LOADED from /home/dev/.config/Lens/config.json
STORE MIGRATION (/home/dev/.config/Lens/lens-cluster-store.json): 2.0.0-beta.2
STORE MIGRATION (/home/dev/.config/Lens/lens-cluster-store.json): 2.4.1
STORE MIGRATION (/home/dev/.config/Lens/lens-cluster-store.json): 2.6.0-beta.2
STORE MIGRATION (/home/dev/.config/Lens/lens-cluster-store.json): 2.6.0-beta.3
STORE MIGRATION (/home/dev/.config/Lens/lens-cluster-store.json): 2.7.0-beta.0
STORE MIGRATION (/home/dev/.config/Lens/lens-cluster-store.json): 2.7.0-beta.1
STORE MIGRATION (/home/dev/.config/Lens/lens-cluster-store.json): 3.6.0-beta.1
STORE MIGRATION (/home/dev/.config/Lens/lens-cluster-store.json): 4.0.1
info: [STORE]: LOADED from /home/dev/.config/Lens/lens-cluster-store.json
info: [STORE]: LOADED from /home/dev/.config/Lens/lens-workspace-store.json
info: [STORE]: LOADED from /home/dev/.config/Lens/lens-extensions.json
info: [STORE]: LOADED from /home/dev/.config/Lens/lens-filesystem-provisioner-store.json
debug: Lookup new free port..
debug: New port found: 43209
info: LensProxy server has started at http://localhost:43209
debug: [EXTENSIONS-LOADER]: load on main
info: [EXTENSION-DISCOVERY] watching extension add/remove in /home/dev/.k8slens/extensions
info: [EXTENSION-DISCOVERY] loading extensions from /home/dev/.config/Lens
debug: [EXTENSION-DISCOVERY]: 6 extensions loaded {"folderPath":"/home/dev/.config/Lens/extensions","extensions":[{"id":"/home/dev/.config/Lens/node_modules/kube-object-event-status/package.json","absolutePath":"/home/dev/.config/Lens/extensions/kube-object-event-status","manifestPath":"/home/dev/.config/Lens/node_modules/kube-object-event-status/package.json","manifest":{"name":"kube-object-event-status","version":"0.1.0","description":"Adds kube object status from events","renderer":"dist/renderer.js","lens":{"metadata":{},"styles":[]},"scripts":{"build":"webpack --config webpack.config.js","dev":"npm run build --watch","test":"echo NO TESTS"},"dependencies":{},"devDependencies":{"@k8slens/extensions":"file:../../src/extensions/npm/extensions","ts-loader":"^8.0.4","typescript":"^4.0.3","webpack":"^4.44.2","mobx":"^5.15.5","react":"^16.13.1"}},"isBundled":true,"isEnabled":true},{"id":"/home/dev/.config/Lens/node_modules/lens-license/package.json","absolutePath":"/home/dev/.config/Lens/extensions/license-menu-item","manifestPath":"/home/dev/.config/Lens/node_modules/lens-license/package.json","manifest":{"name":"lens-license","version":"0.1.0","description":"License menu item","main":"dist/main.js","scripts":{"build":"webpack -p","dev":"webpack --watch","test":"jest --passWithNoTests --env=jsdom src $@"},"dependencies":{},"devDependencies":{"@k8slens/extensions":"file:../../src/extensions/npm/extensions","@types/webpack":"^4.41.17","jest":"^26.6.3","mobx":"^5.15.5","react":"^16.13.1","ts-loader":"^8.0.4","ts-node":"^9.0.0","typescript":"^4.0.3","webpack":"^4.44.2"}},"isBundled":true,"isEnabled":true},{"id":"/home/dev/.config/Lens/node_modules/lens-metrics-cluster-feature/package.json","absolutePath":"/home/dev/.config/Lens/extensions/metrics-cluster-feature","manifestPath":"/home/dev/.config/Lens/node_modules/lens-metrics-cluster-feature/package.json","manifest":{"name":"lens-metrics-cluster-feature","version":"0.1.0","description":"Lens metrics cluster feature","renderer":"dist/renderer.js","lens":{"metadata":{},"styles":[]},"scripts":{"build":"webpack --config webpack.config.js","dev":"npm run build --watch","test":"jest --passWithNoTests --env=jsdom src $@"},"dependencies":{"semver":"^7.3.2"},"devDependencies":{"@k8slens/extensions":"file:../../src/extensions/npm/extensions","jest":"^26.6.3","mobx":"^5.15.5","react":"^16.13.1","ts-loader":"^8.0.4","typescript":"^4.0.3","webpack":"^4.44.2"}},"isBundled":true,"isEnabled":true},{"id":"/home/dev/.config/Lens/node_modules/lens-node-menu/package.json","absolutePath":"/home/dev/.config/Lens/extensions/node-menu","manifestPath":"/home/dev/.config/Lens/node_modules/lens-node-menu/package.json","manifest":{"name":"lens-node-menu","version":"0.1.0","description":"Lens node menu","renderer":"dist/renderer.js","lens":{"metadata":{},"styles":[]},"scripts":{"build":"webpack --config webpack.config.js","dev":"npm run build --watch","test":"jest --passWithNoTests --env=jsdom src $@"},"dependencies":{},"devDependencies":{"@k8slens/extensions":"file:../../src/extensions/npm/extensions","jest":"^26.6.3","mobx":"^5.15.5","react":"^16.13.1","ts-loader":"^8.0.4","typescript":"^4.0.3","webpack":"^4.44.2"}},"isBundled":true,"isEnabled":true},{"id":"/home/dev/.config/Lens/node_modules/lens-pod-menu/package.json","absolutePath":"/home/dev/.config/Lens/extensions/pod-menu","manifestPath":"/home/dev/.config/Lens/node_modules/lens-pod-menu/package.json","manifest":{"name":"lens-pod-menu","version":"0.1.0","description":"Lens pod menu","renderer":"dist/renderer.js","lens":{"metadata":{},"styles":[]},"scripts":{"build":"webpack --config webpack.config.js","dev":"npm run build --watch","test":"jest --passWithNoTests --env=jsdom src $@"},"dependencies":{},"devDependencies":{"@k8slens/extensions":"file:../../src/extensions/npm/extensions","jest":"^26.6.3","mobx":"^5.15.5","react":"^16.13.1","ts-loader":"^8.0.4","typescript":"^4.0.3","webpack":"^4.44.2"}},"isBundled":true,"isEnabled":true},{"id":"/home/dev/.config/Lens/node_modules/lens-telemetry/package.json","absolutePath":"/home/dev/.config/Lens/extensions/telemetry","manifestPath":"/home/dev/.config/Lens/node_modules/lens-telemetry/package.json","manifest":{"name":"lens-telemetry","version":"0.1.0","description":"Lens telemetry","main":"dist/main.js","renderer":"dist/renderer.js","lens":{"metadata":{},"styles":[]},"scripts":{"build":"webpack -p","dev":"webpack --watch","test":"jest --passWithNoTests --env=jsdom src $@"},"dependencies":{},"devDependencies":{"@k8slens/extensions":"file:../../src/extensions/npm/extensions","@types/analytics-node":"^3.1.3","analytics-node":"^3.4.0-beta.3","jest":"^26.6.3","mobx":"^5.15.5","node-machine-id":"^1.1.12","react":"^16.13.1","ts-loader":"^8.0.4","typescript":"^4.0.3","universal-analytics":"^0.4.23","webpack":"^4.44.2"}},"isBundled":true,"isEnabled":true}]}
debug: [EXTENSION-DISCOVERY]: 0 extensions loaded {"folderPath":"/home/dev/.k8slens/extensions","extensions":[]}
info: [EXTENSION-INSTALLER] installing dependencies at /home/dev/.config/Lens
[EXTENSIONS-LOADER]: can't load extension main at /home/dev/.config/Lens/node_modules/lens-license/dist/main.js: Error: Cannot find module '/home/dev/.config/Lens/node_modules/lens-license/dist/main.js'
Require stack:
- /snap/kontena-lens/145/resources/app.asar/static/build/main.js
-  {
  extension: {
    id: '/home/dev/.config/Lens/node_modules/lens-license/package.json',
    absolutePath: '/home/dev/.config/Lens/extensions/license-menu-item',
    manifestPath: '/home/dev/.config/Lens/node_modules/lens-license/package.json',
    manifest: {
      name: 'lens-license',
      version: '0.1.0',
      description: 'License menu item',
      main: 'dist/main.js',
      scripts: [Object],
      dependencies: {},
      devDependencies: [Object]
    },
    isBundled: true,
    isEnabled: true
  }
}
Trace: Error: Cannot find module '/home/dev/.config/Lens/node_modules/lens-license/dist/main.js'
Require stack:
- /snap/kontena-lens/145/resources/app.asar/static/build/main.js
- 
    at Module._resolveFilename (internal/modules/cjs/loader.js:797:17)
    at Function.o._resolveFilename (electron/js2c/browser_init.js:281:681)
    at Module._load (internal/modules/cjs/loader.js:690:27)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Ln.requireExtension (/snap/kontena-lens/145/resources/app.asar/static/build/main.js:1:144605)
    at Object.fireImmediately (/snap/kontena-lens/145/resources/app.asar/static/build/main.js:1:144096)
    at executeAction (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:927:19)
    at Reaction@81 (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:914:16) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/snap/kontena-lens/145/resources/app.asar/static/build/main.js',
    undefined
  ]
}
    at Ln.requireExtension (/snap/kontena-lens/145/resources/app.asar/static/build/main.js:1:144719)
    at Object.fireImmediately (/snap/kontena-lens/145/resources/app.asar/static/build/main.js:1:144096)
    at executeAction (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:927:19)
    at Reaction@81 (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:914:16)
    at reactionRunner (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:2157:13)
    at Reaction.onInvalidate (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:2137:13)
    at Reaction.runReaction (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:1772:26)
    at runReactionsHelper (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:1909:35)
    at reactionScheduler (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:1887:47)
    at runReactions (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:1892:5)
[EXTENSIONS-LOADER]: can't load extension main at /home/dev/.config/Lens/node_modules/lens-telemetry/dist/main.js: Error: Cannot find module '/home/dev/.config/Lens/node_modules/lens-telemetry/dist/main.js'
Require stack:
- /snap/kontena-lens/145/resources/app.asar/static/build/main.js
-  {
  extension: {
    id: '/home/dev/.config/Lens/node_modules/lens-telemetry/package.json',
    absolutePath: '/home/dev/.config/Lens/extensions/telemetry',
    manifestPath: '/home/dev/.config/Lens/node_modules/lens-telemetry/package.json',
    manifest: {
      name: 'lens-telemetry',
      version: '0.1.0',
      description: 'Lens telemetry',
      main: 'dist/main.js',
      renderer: 'dist/renderer.js',
      lens: [Object],
      scripts: [Object],
      dependencies: {},
      devDependencies: [Object]
    },
    isBundled: true,
    isEnabled: true
  }
}
Trace: Error: Cannot find module '/home/dev/.config/Lens/node_modules/lens-telemetry/dist/main.js'
Require stack:
- /snap/kontena-lens/145/resources/app.asar/static/build/main.js
- 
    at Module._resolveFilename (internal/modules/cjs/loader.js:797:17)
    at Function.o._resolveFilename (electron/js2c/browser_init.js:281:681)
    at Module._load (internal/modules/cjs/loader.js:690:27)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Ln.requireExtension (/snap/kontena-lens/145/resources/app.asar/static/build/main.js:1:144605)
    at Object.fireImmediately (/snap/kontena-lens/145/resources/app.asar/static/build/main.js:1:144096)
    at executeAction (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:927:19)
    at Reaction@81 (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:914:16) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/snap/kontena-lens/145/resources/app.asar/static/build/main.js',
    undefined
  ]
}
    at Ln.requireExtension (/snap/kontena-lens/145/resources/app.asar/static/build/main.js:1:144719)
    at Object.fireImmediately (/snap/kontena-lens/145/resources/app.asar/static/build/main.js:1:144096)
    at executeAction (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:927:19)
    at Reaction@81 (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:914:16)
    at reactionRunner (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:2157:13)
    at Reaction.onInvalidate (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:2137:13)
    at Reaction.runReaction (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:1772:26)
    at runReactionsHelper (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:1909:35)
    at reactionScheduler (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:1887:47)
    at runReactions (/snap/kontena-lens/145/resources/app.asar/node_modules/mobx/lib/mobx.js:1892:5)
info: [STORE]: SAVING /home/dev/.config/Lens/config.json
^Cinfo: APP:QUIT
info: SERVICE:QUIT
```

**Comment by panuhorsmalahti** [comment]
4.0.3 will likely have better error handling so we then should be able to see the actual root cause / error for this.

**Comment by buddybu** [comment]
here is a debug log from Windows 10

`console.js:124 info: [STORE]: LOADED from C:\Users\MyUsername\AppData\Roaming\Lens\config.json
console.js:124 info: [STORE]: LOADED from C:\Users\MyUsername\AppData\Roaming\Lens\lens-workspace-store.json
console.js:124 info: [STORE]: LOADED from C:\Users\MyUsername\AppData\Roaming\Lens\lens-cluster-store.json
console.js:124 info: [STORE]: LOADED from C:\Users\MyUsername\AppData\Roaming\Lens\lens-extensions.json
console.js:124 info: [STORE]: LOADED from C:\Users\MyUsername\AppData\Roaming\Lens\lens-filesystem-provisioner-store.json
console.js:124 info: [WORKSPACE-STORE] requesting initial state sync
console.js:124 info: [CLUSTER-STORE] requesting initial state sync
console.js:124 info: [CLUSTER-STORE] start to listen (1)
console.js:124 info: [WORKSPACE-STORE] starting to listen state events
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\kube-object-event-status\dist\renderer.js: Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\kube-object-event-status\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\kube-object-event-status\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:42)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-metrics-cluster-feature\dist\renderer.js: Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-metrics-cluster-feature\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-metrics-cluster-feature\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:42)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-node-menu\dist\renderer.js: Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-node-menu\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-node-menu\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:42)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-pod-menu\dist\renderer.js: Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-pod-menu\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-pod-menu\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:42)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-telemetry\dist\renderer.js: Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-telemetry\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-telemetry\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:42)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2
console.js:124 info: [LENS-VIEW]: init dashboard, clusterId=9279ca41-0071-42df-ae85-d8a5c3cec88c
console.js:124 error: Can't find working API for the Kubernetes resource ingresses
console.js:124 info: [STORE]: LOADED from C:\Users\MyUsername\AppData\Roaming\Lens\config.json
console.js:124 info: [STORE]: LOADED from C:\Users\MyUsername\AppData\Roaming\Lens\lens-workspace-store.json
console.js:124 info: [STORE]: LOADED from C:\Users\MyUsername\AppData\Roaming\Lens\lens-cluster-store.json
console.js:124 info: [STORE]: LOADED from C:\Users\MyUsername\AppData\Roaming\Lens\lens-extensions.json
console.js:124 info: [STORE]: LOADED from C:\Users\MyUsername\AppData\Roaming\Lens\lens-filesystem-provisioner-store.json
console.js:124 info: [WORKSPACE-STORE] requesting initial state sync
console.js:124 info: [CLUSTER-STORE] requesting initial state sync
console.js:124 info: [CLUSTER-STORE] start to listen (5)
console.js:124 info: [WORKSPACE-STORE] starting to listen state events
console.js:124 info: [APP]: Init dashboard, clusterId=9279ca41-0071-42df-ae85-d8a5c3cec88c, frameId=5
console.js:124 info: [LENS-VIEW]: loaded from http://9279ca41-0071-42df-ae85-d8a5c3cec88c.localhost:59666/
9279ca41-0071-42df-ae85-d8a5c3cec88c.localhost:59666/api-kube/apis/networking.k8s.io/v1:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
kube-api.ts:160 e
(anonymous) @ Lens.js:2
9279ca41-0071-42df-ae85-d8a5c3cec88c.localhost:59666/api-kube/apis/apiextensions.k8s.io:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
Lens.js:2 Uncaught (in promise) e
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\kube-object-event-status\dist\renderer.js: Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\kube-object-event-status\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\kube-object-event-status\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:42)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-metrics-cluster-feature\dist\renderer.js: Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-metrics-cluster-feature\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-metrics-cluster-feature\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:42)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-node-menu\dist\renderer.js: Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-node-menu\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-node-menu\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:42)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-pod-menu\dist\renderer.js: Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-pod-menu\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-pod-menu\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:42)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2
extension-loader.ts:282 [EXTENSIONS-LOADER]: can't load extension main at C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-telemetry\dist\renderer.js: Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-telemetry\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init Object
value @ Lens.js:2
extension-loader.ts:283 Error: Cannot find module 'C:\Users\MyUsername\AppData\Roaming\Lens\node_modules\lens-telemetry\dist\renderer.js'
Require stack:
- electron/js2c/renderer_init
    at Module._resolveFilename (internal/modules/cjs/loader.js:797)
    at Function.i._resolveFilename (electron/js2c/renderer_init.js:42)
    at Module._load (internal/modules/cjs/loader.js:690)
    at Function.Module._load (electron/js2c/asar.js:769)
    at Module.require (internal/modules/cjs/loader.js:852)
    at require (internal/modules/cjs/helpers.js:74)
    at e.value (extension-loader.ts:279)
    at extension-loader.ts:245
    at extension-loader.ts:244
    at Be (mobx.module.js:928)
value @ Lens.js:2`

**Comment by panuhorsmalahti** [comment]
Thanks for the logs. I think we need to wait until 4.0.3 to get the actual error.

**Comment by buddybu** [comment]
Until 4.0.3 is released, is there a way to disable auto updates for 3.6.9?  That was the last version that worked for me, but every time I shut down, it automatically upgrades to 4.0.1.  Very frustrating.

**Comment by jhcook** [comment]
I'd like to echo frustration. These capabilities render the product far less useful, and this should be mitigated within a day and corrected in short order -- instead we're a week on. There are people supporting critical systems, and not being able to restore this feature set without using a workaround outside the software's capabilities is pants.

**Comment by evilezh** [comment]
Yep, same situation. Looking to downgrade :)

**Comment by jim-docker** [comment]
We are having trouble reproducing the issue, not everyone is experiencing it. In the mean time I believe the [workaround](https://github.com/lensapp/lens/issues/1690#issuecomment-741636330) suggested by @stijlbreuk above should work, though you may have to add back your clusters after. The corresponding path on Linux is `~/.config/Lens`, and on Windows is  `C:\Users\myusername\AppData\Roaming\Lens`

**Comment by buddybu** [comment]
The workaround specified above does not work for me.  I removed the AppData\Roaming\Lens folder, uninstalled Lens 3.6.9.  Downloaded Lens 4.0.2 and the problem is still present.  My only workaround is to install 3.6.9 and to never shut it off, otherwise it automatically updates to 4.0.2 and the problem. returns.

**Comment by jakolehm** [comment]
I finally managed to reproduce this by doing a fresh install of Lens 4.0.2 in offline mode. It seems that one in-tree extension wants to pull "semver" dependency from network and that causes problems if internet connection is down (or blocked by firewall).

**Comment by jhcook** [comment]
The workaround for macOS users:
1. Use Time Machine so you can restore what the update breaks
2. Remove Lens 4.x
3. Install 3.6.7
4. Do not allow Lens to communicate with GitHub (I use Little Snitch to block it)

You can now start / stop Lens and not worry about an automagic update you can't disable -- feature request?

**Comment by panuhorsmalahti** [comment]
> You can now start / stop Lens and not worry about an automagic update you can't disable -- feature request?

This is tracked in #1768

**Comment by jhcook** [comment]
> I finally managed to reproduce this by doing a fresh install of Lens 4.0.2 in offline mode. It seems that one in-tree extension wants to pull "semver" dependency from network and that causes problems if internet connection is down (or blocked by firewall).

One can never assume the network is reliable and things will work in order as expected. For instance, if I install an update, Little Snitch will block it from outbound comms until I allow it. I imagine that will have a high tendency to disrupt the order in which these things happen. 

That said, I did restore and reinstall a couple times with no success.

**Comment by jakolehm** [comment]
@jhcook agreed, relying to an external service is a bug. It's just really difficult sometimes to spot these kind of bugs.

**Comment by stilettk** [comment]
I see that this issue is "fixed" in v4.0.3, but logs and terminal button are still not available. If this is closed, **what's the solution**? This wasn't a problem before v4.0.0.

Also, I get the following error when opening 4.0.3:
![image](https://user-images.githubusercontent.com/7068016/102454629-62da4d80-404f-11eb-91c7-3e49bf2db852.png)

**Comment by maphysics** [comment]
I ran into this problem upgrading from v3.6.9 to and version 4.  I resolved it with @stijlbreuk's method above.  I removed the `~/Library/Application Support/Lens` and re-installed Lens.  The error message is no longer popping up and the logs button is present.

**Comment by michaelwolz** [comment]
Had the same Issue. I tried completely removing Lens and all related configurations but it did not work out for me on my Ubuntu machine. I manually ran `npm install` in the `.config/Lens` folder which installed all the missing dependencies and now it works just fine for me. Maybe not the best fix but a quick workaround for anyone who still encounters this problem. 

I am on Version 4.1.4 btw.

**Comment by ccrvlh** [comment]
I had 4.2.4 working fine, and after upgrading to 5.0 I lost the logs and terminal button, ironically the ones I used the most in Lens. Deleting everything and reinstalling also doesn't work. Is there a link to download the .dmg for older versions?
I'm looking at the error, in the developer tools, and I have something like:
```
extension-loader.ts:372 [EXTENSIONS-LOADER]: can't load renderer for "lens-telemetry": Error: Cannot find module '/Users/myuser/Library/Application Support/Lens/node_modules/lens-telemetry/dist/renderer.js'
```
Also don't have that file in the pointed path.

I get the errors for:
`lens-metrics-cluster-feature`
`lens-pod-menu`
`lens-node-menu`
`lens-object-event-status`
`lens-survey`

I'm on a mac (with stable connection), also tried `npm install` on `~/Library/Application Support/Lens` and `~/Library/Application Support/Lens/node_modules` (the two places I found a `package.json` file) and nothing.

Manage to make it working by unpacking the `.tgz` file in each `extension/resource` after that, i had a `package` folder. than i moved the `dist` folder to the parent folder (where the app was looking for the `dist`.

**Comment by derekperkins** [comment]
I've got the same problem after upgrading to 5.0.1 on a Mac (Big Sur 11.4)

**Comment by svkempf** [comment]
On my side it's even worse: I had to test out with lots of try and error and installations and uninstalls that 4.0.2 was the last version on my PC where logs were working.
with every newer version the button disappeared. 😢
(running on Windows 10 version 1909)

**Comment by jakolehm** [comment]
5.0.2 includes a fix.

**Comment by svkempf** [comment]
![image](https://user-images.githubusercontent.com/63774617/124550397-dfd4e500-de30-11eb-8f34-63881c44fed5.png)
unfortunately no effect here 🙁

if it helps: I also get this error on start up now
![image](https://user-images.githubusercontent.com/63774617/124550646-43f7a900-de31-11eb-9dd1-4ff65a0c437d.png)

**Comment by jim-docker** [comment]
@svkempf that's very strange. Maybe the Lens user data has been corrupted. You could try renaming `%APPDATA%\Lens\` and then installing 5.0.2 again, just to confirm? If you don't have a lot of Lens configuration saved in `%APPDATA%\Lens\` you can just remove it instead.

**Comment by svkempf** [comment]
I tried this but get the same error on startup and still no logfile button :-(

**Comment by svkempf** [comment]
ah and by the way: Also no terminal button:
![image](https://user-images.githubusercontent.com/63774617/125052575-69d8b400-e0a4-11eb-8a59-fb7baa13ba55.png)

**Comment by ottsion** [comment]
same no terminal and log button

**Comment by ottsion** [comment]
same no terminal and log button

**Comment by ottsion** [comment]
same no terminal and log button
win10

**Comment by jim-docker** [comment]
@svkempf if this is still an issue on Lens 5.1 can you open another issue and add logs for a clean install (remove `%APPDATA%\Lens\` first)?

https://docs.k8slens.dev/main/faq/#how-can-i-see-more-verbose-logs

The Windows app is `C:\Users\your-user-name\AppData\Local\Programs\Lens\Lens.exe`, if you installed it for yourself.

**Comment by msquires** [comment]
I had the same problem on a Mac with Big Sur.  Everything was working fine at one point on 4.x, but then I lost the `pod logs` and `exec to pod` buttons in the UI somewhere after upgrading to v5.x.  I tried all of the following unsuccessfully:
1. Completely uninstalling and re-installing Lens from scratch. 
2. Renaming the ~/Library/Application Support/Lens folder and then uninstalling and re-installing.

What finally worked was running `npm install` in the ~/Library/Application Support/Lens folder.

**Comment by Gowiem** [comment]
@msquires I was experiencing the same on 5.1.3-latest.20210722.1 and your fix did the trick for me. Thanks for the pointer!

**Comment by ConnorDY** [comment]
I tried @msquires fix but it made no difference for me. I have Lens installed via homebrew. No errors in the Developer Tools console either.

**Comment by pcj** [comment]
I had similar issue where somehow file permissions got borked:

```
checkPermissions Missing write access to /Users/i868039/Library/Application Support/Lens/node_modules/lenscloud-lens-extension
npm ERR! code EACCES
npm ERR! syscall access
npm ERR! path /Users/i868039/Library/Application Support/Lens/node_modules/lenscloud-lens-extension
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, access '/Users/i868039/Library/Application Support/Lens/node_modules/lenscloud-lens-extension'
npm ERR!  [Error: EACCES: permission denied, access '/Users/i868039/Library/Application Support/Lens/node_modules/lenscloud-lens-extension'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/Users/i868039/Library/Application Support/Lens/node_modules/lenscloud-lens-extension'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.
```

Fixed by:

```
$ rm -rf node_modules
$ npm i
```

**Comment by ypicard** [comment]
I followed instructions here: https://github.com/alebcay/openlens-node-pod-menu.

**Comment by eucleciojosias** [comment]
@ypicard excellent!

**Comment by john007abhilash** [comment]
> I followed instructions here: https://github.com/alebcay/openlens-node-pod-menu.

Thanks, working now
